internal/handlers/common/filter.go

```diff
@@ -289,17 +299,31 @@ func filterFieldExpr(fieldValue any, expr *types.Document) (bool, error) {
 
 // filterFieldRegex handles {field: /regex/} filter.
 func filterFieldRegex(fieldValue any, regex types.Regex) (bool, error) {
-	s, ok := fieldValue.(string)
-	if !ok {
-		return false, nil
-	}
-
 	re, err := regex.Compile()
 	if err != nil {
 		return false, err
 	}
 
-	return re.MatchString(s), nil
+	switch fieldValue := fieldValue.(type) {
+	case string:
+		return re.MatchString(fieldValue), nil
+
+	case *types.Array:
+		for i := 0; i < fieldValue.Len(); i++ {
+			arrValue := must.NotFail(fieldValue.Get(i)).(any)
+			s, isString := arrValue.(string)
+			if !isString {
+				continue
+			}
+			if re.MatchString(s) == true {
+				return true, nil
+			}
+		}
+	case types.Regex:
```

Comment:
If `regex == regex` ? Don't we expect that it's not equal since we are expecting the comparison towards scalar?